test/RestrictedNORI-transfer.test.ts

```diff
@@ -0,0 +1,531 @@
+          {
+            balance: BigNumber.from(restrictedAmount),
+          }
+        );
+        const scheduleSummaryBeforeTransfer = await rNori.getScheduleSummary(
+          projectId
+        );
+        compareScheduleSummaryStructs(scheduleSummaryBeforeTransfer, {
+          tokenHolders: [supplier],
+        });
+        await rNori
+          .connect(hre.namedSigners.supplier)
+          .safeTransferFrom(
+            supplier,
+            investor1,
+            projectId,
+            restrictedAmount,
+            '0x'
+          );
+        const supplierScheduleDetailAfterTransfer =
+          await rNori.getScheduleDetailForAccount(supplier, projectId);
+        const investor1ScheduleDetailAfterTransfer =
+          await rNori.getScheduleDetailForAccount(investor1, projectId);
+        const scheduleSummaryAfterTransfer = await rNori.getScheduleSummary(
+          projectId
+        );
+
+        const expectedSupplierScheduleDetailAfterTransfer = {
+          tokenHolder: supplier,
+          balance: BigNumber.from(0),
+          claimableAmount: BigNumber.from(0),
+          claimedAmount: BigNumber.from(0),
+          quantityRevoked: BigNumber.from(0),
+          exists: true,
+        };
+        const expectedInvestor1ScheduleDetailAfterTransfer = {
+          tokenHolder: investor1,
+          balance: BigNumber.from(restrictedAmount),
+          claimableAmount: BigNumber.from(0),
+          claimedAmount: BigNumber.from(0),
+          quantityRevoked: BigNumber.from(0),
+          exists: true,
+        };
+        compareScheduleDetailForAddressStructs(
+          supplierScheduleDetailAfterTransfer,
+          expectedSupplierScheduleDetailAfterTransfer
+        );
+        compareScheduleDetailForAddressStructs(
+          investor1ScheduleDetailAfterTransfer,
+          expectedInvestor1ScheduleDetailAfterTransfer
+        );
+        compareScheduleSummaryStructs(scheduleSummaryAfterTransfer, {
+          tokenHolders: [investor1],
+        });
+      });
+    });
+    describe('safeBatchTransferFrom', () => {
+      it('should transfer some of multiple token types to another account', async () => {
+        const removalDataToList = [
+          {
+            amount: 100,
+            vintage: 2018,
+          },
+          {
+            amount: 100,
+            vintage: 2019,
+          },
+        ];
+        const testSetup = await setupTestLocal({});
+        const { rNori, hre } = testSetup;
+        const { listedRemovalIds: listedRemovalIds1, projectId: projectId1 } =
+          await batchMintAndListRemovalsForSale({
+            testSetup,
+            removalDataToList: [removalDataToList[0]],
+          });
+        const {
+          listedRemovalIds: listedRemovalIds2,
+          projectId: projectId2,
+          scheduleStartTime: scheduleStartTime2,
+        } = await batchMintAndListRemovalsForSale({
+          testSetup,
+          projectId: 999_999_999,
```

Comment:
could probably be abstracted better into the removalDataToList or userFixtures. Doesnt need to happen in this pr